docs/4.4/api-reference.md

```diff
@@ -7,61 +7,479 @@ description: The detailed guide to Teleport API
+
+```go
+token, err := client.GetToken(tokenString)
+if err != nil {
+  return err
+}
+```
+
+### Create token
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  Roles: teleport.Roles{teleport.RoleProxy},
+  TTL:   time.Hour,
+})
+if err != nil {
+  return err
+}
 ```
 
-The above sequence should result in three PEM encoded files being generated:
-`auth.crt`, `auth.key`, and `auth.cas` (certificate, private key, and CA certs respectively).
+### Update token
 
-Note: by default, tctl auth sign produces certificates with a relatively short lifetime.
-For production deployments, the --ttl flag can be used to ensure a more practical
-certificate lifetime.
+```go
+// updates the token to be a proxy token
+token.SetRoles(teleport.Roles{teleport.RoleProxy})
+if err := client.UpsertToken(token); err != nil {
+  return err
+}
+```
+
+### Delete token
+
+```go
+if err := client.DeleteToken(tokenString); err != nil {
+  return err
+}
+```
+
+## Cluster Labels
+
+The root cluster in a [trusted cluster](admin-guide.md#adding-nodes-to-the-cluster) can add/update cluster labels on leaf clusters.
+
+### Create a leaf cluster token with Labels
+
+Leaf clusters will inherit the labels of the [join token](trustedclusters.md#join-tokens) used to add them to the trusted cluster. This is the preferred method of managing cluster tokens.
+
+Follow the trusted cluster join token [docs](trustedclusters.md#join-tokens) to use this token to create a leaf cluster using `tctl`.
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can provide 'Token' for a static token name
+  Roles: teleport.Roles{teleport.RoleTrustedCluster},
+  TTL:   time.Hour,
+  Labels: map[string]string{
+    "env": "staging",
+  },
+})
+```
+
+### Update leaf cluster's labels
+
+You can also update a leaf cluster's labels from the root cluster using the auth api if necessary.
+
+```go
+rc, err := client.GetRemoteCluster("leafClusterName")
+if err != nil {
+  return err
+}
+
+md := rc.GetMetadata()
+md.Labels = map[string]string{"env": "prod"}
+rc.SetMetadata(md)
+
+if err = client.UpdateRemoteCluster(ctx, rc); err != nil {
+  return err
+}
+
+```
+
+## Access Workflow
+
+[Access Workflow](enterprise/workflow/index.md) can be used to dynamically approve and deny a user access to local or remote resources in a trusted cluster.
 
-# gRPC APIs
+### Retrieve access requests
+
+```go
+filter := services.AccessRequestFilter{State: services.RequestState_PENDING}
+ars, err := client.GetAccessRequests(ctx, filter)
+if err != nil {
+  return err
+}
+```
+
+### Create access request
+
+```go
+// create a new access request for api-admin to use the admin role in the cluster
+ar, err := services.NewAccessRequest("api-admin", "admin")
+if err != nil {
+  return err
+}
+
+if err = client.CreateAccessRequest(ctx, accessReq); err != nil {
+  return err
+}
+```
 
-## Audit Events API
-Coming Soon
+### Approve access request
+
+```go
```

Comment:
There is an API to add a reason / label back to the audit log. Using plugin data https://github.com/gravitational/teleport-plugins/pull/43/files ( has some pointers )